### PR TITLE
update to flutter 3 and fix lint warnings

### PR DESCRIPTION
**Pull request by noga-dev**
Addresses #117

### CODE REVIEW

**Comment by Perondas** [comment]
I would appreciate this getting merged and published

**Comment by ThexXTURBOXx** [comment]
@LinusU We have some important stuff to fix here again :)

File: lib/flutter_web_auth.dart [hunk 1]
```diff
@@ -29,9 +30,13 @@ class FlutterWebAuth {
...
-  static Future<String> authenticate({required String url, required String callbackUrlScheme, bool? preferEphemeral}) async {
-    WidgetsBinding.instance?.removeObserver(_resumedObserver); // safety measure so we never add this observer twice
-    WidgetsBinding.instance?.addObserver(_resumedObserver);
+  static Future<String> authenticate(
+      {required String url,
+      required String callbackUrlScheme,
+      bool? preferEphemeral}) async {
+    WidgetsBinding.instance.removeObserver(
+        _resumedObserver); // safety measure so we never add this observer twice
+    WidgetsBinding.instance.addObserver(_resumedObserver);
```
Comment (peterhijma) [inline lib/flutter_web_auth.dart:39]: I guess it would help to not change the formatting of the code, so that it's easier to spot the actual changes.

Comment (noga-dev) [reply lib/flutter_web_auth.dart:39]: Understandable. Might as well rip off the band aid now, though.

**Comment by niteshsh4rma** [comment]
When this will be merged?

**Comment by p4-k4** [comment]
Wake up, it's time to merge. My debug console is burning out. 🥱

**Comment by jaumard** [comment]
@LinusU kind reminder to merge this please

**Comment by tulioccalazans** [comment]
@LinusU can you check this PR, please? 
The community likes and is helping this project!

**Comment by nohli** [comment]
@LinusU sorry to disturb you one more time.
Got an idea...since this is a quite popular package, and also a dependency of appwrite (another popular package), it would be great for the community to keep it up to date.
If you like, we could maintain it over at https://github.com/fluttercommunity. Or you could make @ThexXTURBOXx a collaborator. What do you think? 🤗

**Comment by ThexXTURBOXx** [comment]
@nohli Yes, fluttercommunity would be a great idea!
I would also still offer my help in helping maintaining this package :)

**Comment by niteshsh4rma** [comment]
@peterhijma please approve this and merge ASAP, its already too much time since this issue is opened.

**Comment by ThexXTURBOXx** [comment]
I thought about forking this repo and publishing my own fork to pub.dev.
If @LinusU does not reply until Monday, I will be publishing my own version then - of course under compliance with the MIT license.
I will keep you updated here!

**Comment by niteshsh4rma** [comment]
> I thought about forking this repo and publishing my own fork to pub.dev. If @LinusU does not reply until Monday, I will be publishing my own version then - of course under compliance with the MIT license. I will keep you updated here!
> 
> P.S.: If anyone is interested in becoming a co-author, please let me know so that I can add you to my fork as well then :)

Thanks @ThexXTURBOXx , also I'll like to co-author this project.

**Comment by ThexXTURBOXx** [comment]
@niteshsh4rma As soon as I have everything set up, I will let you know!
Thanks for volunteering :)

**Comment by ThexXTURBOXx** [comment]
For follow-ups, see here: https://pub.dev/packages/flutter_web_auth_2

**Pull request closed without merging**